src/ThemeProvider/ThemeProvider.js

```diff
@@ -0,0 +1,46 @@
+/* eslint-disable default-case */
+import PropTypes from 'prop-types';
+import React, { useEffect } from 'react';
+
+const ThemeContext = React.createContext(null);
+
+function ThemeProvider(props) {
+    const { children, theme, disableStyles } = props;
+    useEffect(() => {
+        if (!disableStyles) {
+            //require fonts and icons
+            require('fundamental-styles/dist/fonts.css');
+            require('fundamental-styles/dist/icon.css');
+        }
+    }, []);
+
+    useEffect(() => {
+        if (!disableStyles) {
+            // css variable root declarations
+            console.log(theme)
+            switch (theme) {
+                case 'dark':
+                    require('@sap-theming/theming-base-content/content/Base/baseLib/sap_fiori_3_dark/css_variables.css');
+                    break;
+                case 'light-dark':
+                    require('@sap-theming/theming-base-content/content/Base/baseLib/sap_fiori_3_light_dark/css_variables.css');
+                    break;
+                case 'default':
+                    require('@sap-theming/theming-base-content/content/Base/baseLib/sap_fiori_3/css_variables.css');
```

Comment:
I get where you're going with this and I think there's some merit here.  I am curious if there's a good way to essentially "unrequire" when toggling themes (or anything for that matter).  The `require` statement loads more styles, but it never really unloads any.

Reply:
Yeah, the more research I do, I don't think this is going to be the solution for theming. What do we think about it for fonts and icons though?